src/objectExplorer/objectExplorerUtils.ts

```diff
@@ -25,21 +26,28 @@ export class ObjectExplorerUtils {
     }
 
     public static getNodeUri(node: TreeNodeInfo): string {
-        if (node.nodeType === Constants.disconnectedServerLabel) {
-            return undefined;
+        const profile = <IConnectionProfile>node.connectionCredentials;
+        return ObjectExplorerUtils.getNodeUriFromProfile(profile);
+    }
+
+    public static getNodeUriFromProfile(profile: IConnectionProfile): string {
+        const uri = profile.server + '_' +
```

Comment:
Nit. Using + to concatenate strings is pretty slow (although it doesn't really matter in this case.

Some alternatives are using the new ``` `${}` ``` syntax or `[profile.server, profile.database, profile.user, profile.profileName].join('_');`


Some more reading if you're interested: https://codecalamity.com/stop-using-plus-signs-to-concatenate-strings/